xds/src/main/java/io/grpc/xds/PriorityLoadBalancerProvider.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2020 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.MoreObjects;
+import io.grpc.LoadBalancer;
+import io.grpc.LoadBalancer.Helper;
+import io.grpc.LoadBalancerProvider;
+import io.grpc.NameResolver.ConfigOrError;
+import io.grpc.internal.ServiceConfigUtil.PolicySelection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/** Provider for priority load balancing policy. */
+final class PriorityLoadBalancerProvider extends LoadBalancerProvider {
+
+  @Override
+  public boolean isAvailable() {
+    return true;
+  }
+
+  @Override
+  public int getPriority() {
+    return 5;
+  }
+
+  @Override
+  public String getPolicyName() {
+    return "priority_experimental";
+  }
+
+  @Override
+  public LoadBalancer newLoadBalancer(Helper helper) {
+    return new PriorityLoadBalancer(helper);
+  }
+
+  @Override
+  public ConfigOrError parseLoadBalancingPolicyConfig(Map<String, ?> rawConfig) {
+    throw new UnsupportedOperationException();
+  }
+
+  static final class PriorityLbConfig {
+    final Map<String, PolicySelection> childConfigs;
+    final List<String> priorities;
+
+    PriorityLbConfig(Map<String, PolicySelection> childConfigs, List<String> priorities) {
+      this.childConfigs = Collections.unmodifiableMap(checkNotNull(childConfigs, "childConfigs"));
+      this.priorities = Collections.unmodifiableList(checkNotNull(priorities, "priorities"));
+      checkArgument(!priorities.isEmpty(), "priority list is empty");
+      checkArgument(
+          childConfigs.keySet().containsAll(priorities),
```

Comment:
Maybe also check the other direction to ensure exact 1:1 mapping?

Reply:
Yes, but sometime it's easy to miss some non-essential checks. Another check is that the list should not have duplicate names.

Reply:
An implementation can do anything (even halt the program) if the input does not meet the precondition. Checks are just performing quick screenings to the input and fail uniformly if precondition is not met. If checks are expensive (e.g., no duplicates), we should not do it. It's caller's responsibility to ensure the precondition of the callee method is met.